javascript-carousel/main.js

```diff
@@ -0,0 +1,133 @@
+// DOM queries and other variables
+var imageList = document.querySelectorAll('img');
+var buttonsList = document.querySelectorAll('.button-image');
+var timerID = null;
+
+// image change every so and so seconds
+function imageChangeWithInterval() {
+  imageChangeForward();
+  timerID = setTimeout(imageChangeWithInterval, 1300);
+}
+// imageChangeWithInterval();
+
+var activeIndex = 0;
+
+function displayImage(index) {
+  for (var i = 0; i < imageList.length; i++) {
+    if (i !== index) {
+      imageList[i].classList.add('hidden');
+      buttonsList[i].classList.remove('current-button');
+    }
+    imageList[index].classList.remove('hidden');
+    buttonsList[index].classList.add('current-button');
+  }
+  activeIndex = index;
+}
+
+function getNextIndex() {
+  if (activeIndex === imageList.length - 1) {
+    return 0;
+  }
+  return activeIndex + 1;
+}
+
+function getPreviousIndex() {
+  if (activeIndex === 0) {
+    return imageList.length - 1;
+  }
+  return activeIndex - 1;
+}
+
+// image changes going forward
+function imageChangeForward() {
+  var currentImage = document.querySelector('.current-pokemon');
+  var currentButton = document.querySelector('.current-button');
+
+  var nextImage = currentImage.nextElementSibling;
+  var nextButton = currentButton.nextElementSibling;
+
+  if (currentImage.id === 'jigglypuff' || currentButton.id === 'lastbutton') {
+    nextImage = document.querySelector('#bulbasaur');
+    nextButton = document.querySelector('#firstbutton');
+  }
+
+  currentImage.classList.remove('current-pokemon');
+  currentImage.classList.add('hidden');
+  nextImage.classList.remove('hidden');
+  nextImage.classList.add('current-pokemon');
+
+  currentButton.classList.remove('current-button');
+  nextButton.classList.add('current-button');
```

Comment:
Ok, in here you can get the next index and display the proper image with your functions.